### ISSUE TITLE
[BUG]: Fail reasons are not correctly mapping to Policies of Violation, Needs Review, and Recommendations.

### ISSUE THREAD

**Issue by philljenkins** [IBMa/equal-access#1731]
### Project

accessibility-checker-engine

### Browser

Chrome, Firefox

### Operating system

MacOS, Windows

### Description

Rules with multiple fail reasons are not correctly displaying the correct policy (Violation, Needs review, Recommendation) in the Help pages.  

For example, the new WCAG 2.2 rule: [target_spacing_sufficient.ts](https://github.com/IBMa/equal-access/blob/dev-1674/accessibility-checker-engine/src/v4/rules/target_spacing_sufficient.ts) should map to 
`Violation: "violation_spacing"` via the `RuleFail("violation_spacing",` mapping
`Needs review: "potential_overlap"` via the `RulePotential("potential_overlap",` mapping
`Recommendation: "recommendation_inline"` via the `RulePass("recommendation_inline",` mapping

But, "all" the fail codes seem to only map to `Recommendation`

### Steps to reproduce

1. run the test cases in [target_spacing_sufficient_ruleunit](https://github.com/IBMa/equal-access/tree/dev-1674/accessibility-checker-engine/test/v2/checker/accessibility/rules/target_spacing_sufficient_ruleunit)
2. display the Learn more ... Help 

For example. the following mapping is incorrect, it should be a Violation (not a Recommendation)
![image](https://github.com/IBMa/equal-access/assets/16193609/b5650778-dbfd-4642-8fae-3106ac18721e)

**Comment by shunguoy** [comment]
PR https://github.com/IBMa/equal-access/pull/1745 for review by @tombrunet

**Comment by ErickRenteria** [comment]
Merged
